homeassistant/components/filesize/config_flow.py

```diff
@@ -0,0 +1,80 @@
+"""The filesize config flow."""
+from __future__ import annotations
+
+import logging
+import pathlib
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow
+from homeassistant.const import CONF_FILE_PATH
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DOMAIN
+
+DATA_SCHEMA = vol.Schema({vol.Required(CONF_FILE_PATH): str})
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def validate_path(hass: HomeAssistant, path: str) -> pathlib.Path:
+    """Validate path."""
+    try:
+        get_path = pathlib.Path(path)
+    except OSError as error:
+        _LOGGER.error("Can not access file %s, error %s", path, error)
+        raise NotValidError from error
+
+    if not hass.config.is_allowed_path(path):
+        _LOGGER.error("Filepath %s is not valid or allowed", path)
+        raise NotAllowedError
+
+    return get_path
+
+
+class FilesizeConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Config flow for Filesize."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initialized by the user."""
+        errors: dict[str, Any] = {}
+
+        if user_input is not None:
+            try:
+                get_path = validate_path(self.hass, user_input[CONF_FILE_PATH])
+                fullpath = str(get_path.absolute())
```

Comment:
```suggestion
```

Reply:
I don't think this can throw `NotValidError` or `NotAllowedError` so lets move it into the else